### PR TITLE
cleaned up burial and organic matter remin, added options to toggle/scale burial

### PR DESCRIPTION
**Pull request by charliestock**
This pull request responds to issue #70.  It adds an option to toggle and scale burial with a parameter called "scale_burial".  The default value is 0 to match prior simulations that had burial turned off.  Changing the value to 1.0 turns on the full burial amount, and intermediate values produce intermediate levels of burial.

One can also modulate burial with the parameter z_burial, which provides a depth scale over which burial is ramped up to its full value.  This parameter is useful to avoid excessive burial in nearshore regions.  The default value of this is 10m.

In prior versions of the code, z_burial was also used to scale benthic denitrification in nearshore waters.  I changed this to z_denit to avoid any confusion.  The default value for this is also 10m, which is what we were using in prior versions of the code.

The section of the code handling the transformations of particulate organic matter in the benthos was also cleaned up for this pull request.  This includes the addition of substantial comments.  This addition should not change answers, but it should ease exploration of the impact of burial on the hypoxic zone in association with issue #70.

### CODE REVIEW

**Comment by yichengt900** [comment]
@charliestock,

Thanks for adding the new `scale_burial` option. Below, please find the test results with `scale_burial = 1.0` (red line, `z_burial = 10m` for all the tests). As we expected, the hypoxic zone has shrunk and is now pretty close to the baseline (results from Andrew's paper).

hypoxic_area_from_monthly

**Comment by charliestock** [comment]
Great news @yichengt900.  Would it be possible to do a short runs with z_burial = 20m and z_burial = 30m to quantify the general sensitivity.  I would think it would then be safe to pull this in.

**Comment by yichengt900** [comment]
@charliestock, definitely. Below, please find the additional sensitivity tests for `z_burial=20m` (magenta) and `z_burial=30m` (yellow). As we expected, `z_burial` can reduce excessive burial in shallow waters (e.g., nearshore regions) and, as a consequence, increase the hypoxic zone.
 
hypoxic_area_from_monthly

**Comment by charliestock** [comment]
Thanks @yichengt900.  This really helps to understand the burial sensitivity, though it seems like we would have more work to do to fully understand and calibrate the hypoxic zone dynamics.  The challenge seems to be pushing the initiation earlier about 2 weeks to a month earlier in the year while also reducing it earlier.  It will probably take a more focused look at the phys and BGC dynamics to trim up the dynamics further.   However, even the "no burial" case gives a hypoxic zone of reasonable size.  If I had to choose one of the options as the "best", it looks like scale_burial = 1 and z_burial = 20m.

We could perhaps do a quick complementary pair of simulations with z_denit = 0m and z_denit = 20m to assess this sensitivity, branching these from the scale_burial=1/z_burial = 20m case.  Lower z_denit should reduce hypoxia by allocating more sedimentary remineralization to anaerobic pathways, and higher z_denit should do the opposite.  I'd be particularly curious whether z_denit = 20m shifts the development of hypoxia earlier in the year.  @yichengt900, could I cajole you into running two more quick sensitivity tests?

Once we have these in hand, perhaps @andrew-c-ross could start a longer simulation to understand the implications of any burial/denitrification changes done to improve the hypoxic zone on other aspects of the simulation?  What do you guys think?

**Review by yichengt900 — APPROVED** [review]
@charliestock, the code changes and comments look good to me. I am still waiting for the `z_denit` sensitivity test results (gaea is down today). I approve this PR and will merge it once we get the final results from the `z_denit` tests.

**Comment by charliestock** [comment]
Sounds good Yi-Cheng, thank you!

**Comment by yichengt900** [comment]
@charliestock, below is the plot for the two additional sensitivity experiments (`z_denit=0m` and `z_denit=20m` with `scale_burial = 1` and `z_burial = 20m`). It appears that `z_denit` does not significantly affect the peak, although there is still a delay of a few weeks. However, it does slightly alter the shape after the July peak.

hypoxic_area_from_monthly

**Comment by andrew-c-ross** [comment]
@charliestock @yichengt900 I'd be happy to run a longer simulation if we've settled on a set of parameters. 

I've also been working on a diagnostic for bottom dissolved oxygen at the [A01 mooring in Massachusetts Bay](http://gyre.umeoce.maine.edu/data/gomoos/buoy/html/A01.html). This could be a test for whether this PR has any relevant effects outside of the Mississippi area. The simulation I ran in April looks pretty decent, with a slight high oxygen bias mainly from missing a few of the observed low extremes. Most of the observed daily and monthly variability is not captured, though in addition to being a very nearshore point I think there are relevant processes like wastewater discharge that we don't model. 

bottom_do_obs

File: generic_tracers/generic_COBALT.F90 [hunk 1]
```diff
@@ -4144,45 +4152,70 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
...
-             !     cobalt%zt(i,j,k) / (cobalt%z_burial + cobalt%zt(i,j,k))
-             cobalt%frac_burial(i,j) = 0.0
+             cobalt%frac_burial(i,j) = 0.013 + 0.53*fpoc_btm**2.0/((7.0+fpoc_btm)**2.0) * &
+                  cobalt%zt(i,j,k) / (cobalt%z_burial + cobalt%zt(i,j,k))
+             cobalt%frac_burial(i,j) = cobalt%scale_burial*cobalt%frac_burial(i,j)
              cobalt%fn_burial(i,j) = cobalt%frac_burial(i,j)*cobalt%fntot_btm(i,j)
              cobalt%fp_burial(i,j) = cobalt%frac_burial(i,j)*cobalt%fptot_btm(i,j)
 
-             ! Denitrification
+             ! Denitrification follows Middelberg et al., 1996. Denitrification in marine sediments: a modeling study
```
Comment (jessluo) [inline generic_tracers/generic_COBALT.F90]: quick spelling correction: should be Middelburg (with a u, not an e).

File: generic_tracers/generic_COBALT.F90 [hunk 2]
```diff
@@ -4144,45 +4152,70 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
...
+             ! Global Biogeochemical Cycles 10(4).  pp. 661-673.  https://doi.org/10.1029/96GB02562. COBALT uses the  
+             ! carbon flux-based relationship based on Middelberg's first extraction of his metamodel (the first
```
Comment (jessluo) [inline generic_tracers/generic_COBALT.F90]: same here, "Middelburg"

File: generic_tracers/generic_COBALT.F90 [hunk 3]
```diff
@@ -4144,45 +4152,70 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
...
+             ! equation in Section 3.4 of the paper).  This relationship requires a flux to the benthis in micromoles C
```
Comment (jessluo) [inline generic_tracers/generic_COBALT.F90]: "benthos"

File: generic_tracers/generic_COBALT.F90 [hunk 4]
```diff
@@ -4144,45 +4152,70 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
...
+             ! cm-2 day-1.  This means that fpoc_btm defined for the burial calculation above must be multiplied by:
+             ! 
+             ! 1e3 micromoles/millimole*1e-4 cm2/m2 = 0.1
+             ! 
+             ! to get the proper units.  The Middelberg relationship yields a rate at which arriving particulate organic
```
Comment (jessluo) [inline generic_tracers/generic_COBALT.F90]: "Middelburg"

File: generic_tracers/generic_COBALT.F90 [hunk 5]
```diff
@@ -4144,45 +4152,70 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
...
+             ! 
+             ! c_2_n*sperd*1e6 micromoles/mole*1e-4 cm2/m2 = c_2_n*sperd*100
+             !
+             ! The nitrate demand associated with this denitrification (fno3denit_sed) is obtained by multiplying the
+             ! resulting value by the moles of NO3 required to denitrify each mole of organic N (n_2_n_denit).
              !
-             ! Middelberg, log10(fpoc_bottom) in micromoles C cm-2 day-1 for denitrification relationship
-             ! cap at 43.0 to prevent anomalous extrapolation of the relationship
+             ! A number of limiters are applied to support global application.  First, the C flux used in the  
+             ! Middelberg relationship is capped at 43.0 micromoles C cm-2 day-1 to avoid anomalous extrapolation.
```
Comment (jessluo) [inline generic_tracers/generic_COBALT.F90]: "Middelburg"

File: generic_tracers/generic_COBALT.F90 [hunk 6]
```diff
@@ -4144,45 +4152,70 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
...
+             ! Second, denitrification is slowed when bottom nitrate is low by a) scaling rates with a nitrate
+             ! half-saturation constant with (k_no3_denit), b) preventing the exhaustion of bottom nitrate over
+             ! single time step, and c) limiting the total amount of organic carbon denitrified to that arriving at
+             ! the sediment minus that which was buried. Finally, to prevent excessive denitrification in very shallow
+             ! areas, a depth scale (z_denit) was included to ramp up rates to full Middelberg values only in deeper
```
Comment (jessluo) [inline generic_tracers/generic_COBALT.F90]: "Middelburg"

**Review by jessluo — COMMENTED** [review]
A few spelling corrections. Otherwise looks good to me!

**Comment by yichengt900** [comment]
@jessluo, thanks for the comments! I've fixed the typos. @charliestock, please feel free to chime in if you think this PR is ready to go.

**Comment by andrew-c-ross** [comment]
I will run NWA12 with 10m/10m and 20m/20m

**Comment by yichengt900** [comment]
@andrew-c-ross and @amoebaliz,

I have merged this PR into the dev/cefi branch. I also updated our CEFI-regional-MOM6 repository, so you can use it directly for the scal_burial sensitivity experiments. Similar to MOM6, we can now define and override the default parameters using `COBALT_input` and `COBALT_override` respectively:
```
do_case2_mod = True       
scale_burial = 1.0
z_burial = 20.0
z_denit = 20.0
```

**Comment by andrew-c-ross** [comment]
@charliestock @yichengt900 just to finally follow up, I ran this with z_burial = z_denit = 10 m and 20 m. 20 m produces much better results for Gulf of Mexico: 
hypoxic_area_from_monthly_2024_07_20m
compared to 10 m:
hypoxic_area_from_monthly_2024_07_10m

**Comment by andrew-c-ross** [comment]
20 m overestimates oxygen in Massachusetts Bay, however. 10 m is similar. scale_burial = 0 looks better. 
bottom_do_obs

**Comment by yichengt900** [comment]
Thanks, @andrew-c-ross. Do you think a spatially varying scale_burial/z_burial would be helpful?

**Comment by andrew-c-ross** [comment]
I'm running a simulation to see what difference scale_burial makes. Generally I'm not super concerned about it, since a lot of the overestimation is due to the extreme observed lows around 2016 and 2020 (and I think I read 2020 was from an algal bloom).

**Comment by charliestock** [comment]
Hi @andrew-c-ross, as we discussed last Tuesday, the 20m z_burial looks great.  It was good to see that it cleaned up the timing issues evident in earlier experiments along with providing a great way to modulate mean hypoxia levels.  I also share your perspective on Mass Bay.  The simulation looks pretty darn good for about 80-90% of the years before.  It will be good to look at what dynamics might be missing in the extreme years for future efforts.  Note that we also don't include the Boston sewage outflow, which may play a role in Mass Bay stations.  I'll be curious what our Woods Hole colleagues think!

**Pull request merged**